apps/aecore/lib/aecore/miner/worker.ex

```diff
@@ -321,7 +319,7 @@ defmodule Aecore.Miner.Worker do
     txs_list = get_pool_values()
     ordered_txs_list = Enum.sort(txs_list, fn tx1, tx2 -> tx1.data.nonce < tx2.data.nonce end)
 
-    candidate_height = top_height
+    candidate_height = Chain.top_height()
```

Comment:
What if we want to mine on top of an old block(we just now received a malicious block)?